src/odemis/acq/move.py

```diff
@@ -0,0 +1,314 @@
+    :param rx: (float) rotation movement in x axis
+    :param rz: (float) rotation movement in z axis
+    :return (CancellableFuture -> None): cancellable future of the move to observe the progress, and control the
+   raise ValueError exception
+    """
+    # Get the stage and focus components from the backend components
+    stage = model.getComponent(role='stage')
+    focus = model.getComponent(role='focus')
+
+    f = model.CancellableFuture()
+    f.task_canceller = _cancelCryoMoveSample
+    f._task_state = RUNNING
+    f._task_lock = threading.Lock()
+    f._running_subf = model.InstantaneousFuture()
+    # Run in separate thread
+    executeAsyncTask(f, _doCryoTiltSample, args=(f, stage, focus, rx, rz,))
+    return f
+
+
+def _cancelCryoMoveSample(future):
+    """
+    Canceller of _doCryoTiltSample and _doCryoLoadSample tasks
+    """
+    logging.debug("Cancelling CryoMoveSample...")
+
+    with future._task_lock:
+        if future._task_state == FINISHED:
+            return False
+        future._task_state = CANCELLED
+        future._running_subf.cancel()
+        logging.debug("CryoMoveSample cancellation requested.")
+
+    return True
+
+
+def _doCryoLoadSample(future, stage, focus, target):
+    """
+    Do the actual switching procedure for the Cryo sample stage between loading and imaging
+    :param future: cancellable future of the move
+    :param stage: sample stage that's being controlled
+    :param focus: focus for optical lens
+    :param target: target position either one of the constants LOADING or IMAGING
+    """
+    try:
+        stage_active = stage.getMetadata()[model.MD_FAV_POS_ACTIVE]
+        stage_deactive = stage.getMetadata()[model.MD_FAV_POS_DEACTIVE]
+        focus_deactive = focus.getMetadata()[model.MD_FAV_POS_DEACTIVE]
+        # Create axis->pos dict from target position given smaller number of axes
+        filter_dict = lambda keys, d: {key: d[key] for key in keys}
+
+        if target == LOADING:
+            # Check that the stage X,Y are within the limits
+            for axis in {'x', 'y', 'z'}:
+                if not _isInRange(stage, axis):
+                    logging.warning("Axis %s position is out of active range.", axis)
+            # Check that current position is near the ACTIVE to DEACTIVE line
+            # (ie, in case it was previously stopped half-way through loading/unloading)
+            if getLoadingProgress(stage.position.value, stage_active, stage_deactive) is None:
+                logging.warning("Stage position is not near the ACTIVE to DEACTIVE line.")
+
+            if abs(stage_deactive['rx']) > ATOL_ROTATION_POS:
+                raise ValueError("Absolute value of rx for FAV_POS_DEACTIVE is greater than {}".format(ATOL_ROTATION_POS))
+
+            # Create an ordered sub moves list to perform the loading move
+            sub_moves = [
+                (focus, focus_deactive),
+                (stage, filter_dict({'z'}, stage_deactive)),
```

Comment:
Sorry, now that I think of it, I think it's probably safer to move the "z" near the loading position (because there are less things that could collide there). So you should do this move last, and for the "IMAGING", you should do this move first.
I'll update the spec (for us to remember).

Reply:
Makes sense, I will do as such.